### PR TITLE
Add ARC support

### PR DESCRIPTION
**Pull request by abrodkin**
This adds both ARCv2 (EM & HS families) as well as 64-bit ARCv3 support.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -56,8 +56,8 @@ have_block = have_system or have_tools
 python = import('python').find_installation()
 
 supported_oses = ['windows', 'freebsd', 'netbsd', 'openbsd', 'darwin', 'sunos', 'linux']
-supported_cpus = ['ppc', 'ppc64', 's390x', 'riscv32', 'riscv64', 'x86', 'x86_64',
-  'arm', 'aarch64', 'mips', 'mips64', 'sparc', 'sparc64']
+supported_cpus = ['ppc', 'ppc64', 's390x', 'sparc64', 'riscv32', 'riscv64', 'x86', 'x86_64',
```
Comment (galak) [inline meson.build:59]: sparc64 is getting duplicated here and end of this list.

**Comment by abrodkin** [comment]
Right, was fixed a while ago by https://github.com/qemu/qemu/commit/6125673eaf711247405e796604204bb7d74090f4
Thanks for the hint!

**Comment by galak** [comment]
> This adds both ARCv2 (EM & HS families) as well as 64-bit ARCv3 support.

Is there a new arc64-softmmu target?

**Comment by abrodkin** [comment]
Indeed, for ARC64 we build a separate binary which is exactly `qemu-system-arc64`.
Configure with `./configure --target-list=arc-softmmu,arc64-softmmu`.

**Comment by galak** [comment]
> Indeed, for ARC64 we build a separate binary which is exactly `qemu-system-arc64`.
> Configure with `./configure --target-list=arc-softmmu,arc64-softmmu`.

I've hackishly pulled this in over at:

https://github.com/zephyrproject-rtos/sdk-ng/pull/330

so we can get a build to test out.

**Comment by galak** [comment]
looks like the some of what we had in the current SDK for arc has gone away.  Looking at file diffs in hw/arc/ and how sim-hs.c and nsim.c are missing.

**Comment by abrodkin** [comment]
@galak right, on upstreaming we switched `simhs` to `virt` and `nsim` is gone at all - that's all exepcted.
And we implement that `simhs` -> `virt` switch in https://github.com/zephyrproject-rtos/zephyr/pull/34393 here: 
https://github.com/zephyrproject-rtos/zephyr/pull/34393/files#diff-07aa04e54a3e51a12719e53aafc6805f88ce712b4a0fa473c1602b106c6a9987R18

**Comment by stephanosio** [comment]
Closing obsolete PR.

**Pull request closed without merging**